scipy/optimize/optimize.py

```diff
@@ -2116,14 +2123,15 @@ def _minimize_scalar_golden(func, brack=None, args=(),
             f2 = f1
             f1 = func(*((x1,) + args))
         funcalls += 1
+        nit += 1
     if (f1 < f2):
         xmin = x1
         fval = f1
     else:
         xmin = x2
         fval = f2
 
-    return OptimizeResult(fun=fval, nfev=funcalls, x=xmin)
+    return OptimizeResult(fun=fval, nfev=funcalls, x=xmin, nit=nit)
```

Comment:
Now that the function can fail to return a converged result, the return status should also indicate that, see `_minimize_scalar_brent`.

It's not possible to return the error status from `golden` due to backward compatibility.
That is probably not a problem, provided we use a big enough default value for `maxiter`.